File: brickport-infrastructure-services-domain/src/queries/game-queries.cs
using System;
using System.Collections.Generic;
using System.Threading.Tasks;
using BrickPort.Services.Queries;

namespace BrickPort.Infrastructure.Services.Domain.Queries
{
    public class DomainGameQueries : IGameQueries
    {
        public Task<IEnumerable<GameSummary>> SummaryAsync()
        {
            throw new NotImplementedException();
        }

        public Task<GameSummary> SummaryAsync(string id)
        {
            throw new NotImplementedException();
        }

        public Task<IEnumerable<GameSummary>> SummaryByDateAsync(DateTime? startUtc, DateTime? endUtc)
        {
            throw new NotImplementedException();
        }
    }
}

File: brickport-ui/src/infrastructure/brickport-services/brickport-services-api/queries/game-queries.ts
import { IGameQueries, IGameSummary } from "../../../../brickport-services/queries/game-queries"
import { BrickportApiClient } from "../api-client";

export class ApiGameQueries implements IGameQueries {
    
    private readonly _apiClient: BrickportApiClient;
    
    constructor(endpoint: string) {
        this._apiClient = new BrickportApiClient(endpoint, true);
    }

    summaryAsync(): Promise<IGameSummary[]>;    
    summaryAsync(id: string): Promise<IGameSummary>;
    summaryAsync(id?: string): Promise<IGameSummary | IGameSummary[]> {
        if (id == null) {
            return this._apiClient.get<IGameSummary[]>();
        }
        return this._apiClient.get<IGameSummary>(`/${id}`);
    }

    summaryByDateAsync(startUtc: string | null, endUtc: string | null): Promise<IGameSummary[]> {
        throw new Error("Method not implemented.");
    }


}